rust/connlib/tunnel/src/device_channel/tun_windows.rs

```diff
@@ -254,29 +254,29 @@ fn start_recv_thread(
         .name("Firezone wintun worker".into())
         .spawn(move || {
             loop {
-                match session.receive_blocking() {
-                    Ok(pkt) => {
-                        // Use `blocking_send` so that if connlib is behind by a few packets,
-                        // Wintun will queue up new packets in its ring buffer while we
-                        // wait for our MPSC channel to clear.
-                        match packet_tx.blocking_send(pkt) {
-                            Ok(()) => {}
-                            Err(_) => {
-                                tracing::info!(
-                                    "Stopping outbound worker thread because the packet channel closed"
-                                );
-                                break;
-                            }
-                        }
-                    }
+                let pkt = match session.receive_blocking() {
+                    Ok(pkt) => pkt,
                     Err(wintun::Error::ShuttingDown) => {
-                        tracing::info!("Stopping outbound worker thread because Wintun is shutting down");
+                        tracing::info!(
+                            "Stopping outbound worker thread because Wintun is shutting down"
+                        );
                         break;
                     }
                     Err(e) => {
                         tracing::error!("wintun::Session::receive_blocking: {e:#?}");
                         break;
                     }
+                };
+
+                match packet_tx.try_send(pkt) {
+                    Ok(()) => {}
+                    Err(TrySendError::Closed(_)) => {
+                        // This is redundant since we aren't using
+                        // `blocking_send` anymore but it's defense in depth.
+                        tracing::info!("Closing worker thread because packet channel closed");
+                        break;
+                    }
+                    Err(TrySendError::Full(_)) => {} // Just drop the packet, it's IP
```

Comment:
Yeah this block will need to change back to `blocking_send` since it was partially reverted in #5598

And it might change again in the future because I'd like to know if we drop packets. With `blocking_send` they'll drop inside Wintun silently. With `try_send` and a bigger MPSC buffer, we could know if we drop them, although they could still drop silently inside Wintun